src/utils.js

```diff
@@ -168,15 +168,17 @@ export function getDefaultFormState(_schema, formData, definitions = {}) {
   }
   const schema = retrieveSchema(_schema, definitions);
   const defaults = computeDefaults(schema, _schema.default, definitions);
+  let result;
   if (typeof formData === "undefined") {
     // No form data? Use schema defaults.
-    return defaults;
-  }
-  if (isObject(formData)) {
+    result = defaults;
+  } else if (isObject(formData)) {
     // Override schema defaults with form data.
-    return mergeObjects(defaults, formData);
+    result = mergeObjects(defaults, formData);
+  } else {
+    result = formData || defaults;
   }
-  return formData || defaults;
+  return trimFormData(result);
```

Comment:
This means that we trim defaults, which feels like it could be a bit surprising?